src/send-commercial-metrics.ts

```diff
@@ -75,15 +77,18 @@ const transformToObjectEntries = (
 	return Object.entries(eventTimerProperties);
 };
 
-const filterUndefinedProperties = <T>(
-	transformedProperties: Array<[string, T | undefined]>,
-): Array<[string, T]> =>
-	transformedProperties.reduce<Array<[string, T]>>((acc, [key, value]) => {
-		if (typeof value !== 'undefined') {
-			acc.push([key, value]);
-		}
-		return acc;
-	}, []);
+const filterUndefinedProperties = (
```

Comment:
Just wondering if we could simplify this to a regular `array.filter()` ?

In which case I think the comparison would simplify down to `!!value` ?

Reply:
+1 for `filter` if this is possible but I think we'd still need to test for `value !== 'undefined'`. I think we want to preserve the behavior of sending metrics whose values are defined but falsy like 0, null, empty string.

Reply:
Ah yes you're right. Good old falsy!